### DISCUSSION TITLE
Potential Issue with Next.js App Router: Dynamic Segment Staleness

### DISCUSSION THREAD

**Discussion by ellemedit** [vercel/next.js#52370] [App Router]
I am encountering a problem with the Next.js application router, specifically with dynamic pages.

The issue seems to be that router.push doesn't refresh the page if it's cached. This behavior persists even in Next.js version 13.4.9. You can replicate this issue via the following link: https://github.com/Beingbook/next-link-app-dir-bug (forked from: https://github.com/vercel/next.js/issues/49300)

Additional Replication Steps:

1. Navigate directly to: http://localhost:3000/dynamic
2. Update the text in the input field and click 'Submit'. This will save your input data to the 
body.json file in the root directory.
3. Click on 'Home' to navigate back.
4. Now, click on 'Dynamic' to return to the previous page. You should see the updated value from the input you provided earlier.
5. Proceed to update the value again to something different and click 'Submit'. This saves the new input to body.json.
6. Click on 'Home' once more.
7. Click on 'Dynamic' again.
8. At this point, you may notice that the value has not been updated, even though the most recent input has indeed been saved to body.json. This seems to suggest that the page is not being refreshed properly, despite the data being updated in the back end.

This additional detail illustrates the potential UX issue I mentioned previously, and further supports the need for resolution at the framework level.

I wanted to inquire if this is an intended feature or an inadvertent bug. This could potentially lead to significant user experience degradation, as users may frequently encounter stale content.

A current workaround is to utilize router.refresh() before router.push('...'). While this does temporarily rectify the issue, I believe that a more appropriate solution should be implemented at the framework level.

Could you please provide clarification or consider addressing this problem in future updates if it's not part of the intended behavior? Thank you.

**Comment by icyJoseph** [comment]
I've run into this issue. As far as I know, today, you need to revalidate the path when you "submit" the button. To do this you can use a `server action` and call revalidate inside of it.

- https://github.com/vercel/next.js/discussions/51612#discussioncomment-6248259
- https://github.com/vercel/next.js/discussions/51488#discussioncomment-6216934
